mantle/kola/tests/upgrade/basic.go

```diff
@@ -220,6 +227,40 @@ func (g *Graph) addUpdate(c cluster.TestCluster, m platform.Machine, version, pa
 	g.sync(c, m)
 }
 
+// XXX: consider making this distinction part of FCOS itself?
+func onProdStream(c cluster.TestCluster, d *util.RpmOstreeDeployment) bool {
+	switch d.BaseCommitMeta.FedoraCoreOSStream {
+	case "":
+		c.Fatalf("missing fedora-coreos.stream metadata key")
+	case "stable", "testing", "next":
+		return true
+	}
+
+	return false
+}
+
+func isDevBuild(c cluster.TestCluster, d *util.RpmOstreeDeployment) bool {
+	return strings.Contains(d.Version, "dev")
+}
+
+// On production streams, we never want to explicitly add `updates.enabled =
+// true` since that should be the default. On all other streams, we do.
+func handleZincatiEnabledConfig(c cluster.TestCluster, m platform.Machine) {
+	c.MustSSH(m, "sudo rm -rf /etc/zincati/config.d/99-enabled.toml")
```

Comment:
(I got lost here, I feel like I'm missing something) Shouldn't this be `90-disable-auto-updates.toml` from https://github.com/coreos/mantle/pull/1018/files?

Reply:
Additionally, isn't this something that should be done before rebooting so that it takes effect directly on the next boot on the new stream (without races)?

Reply:
Yeah, I think you're right. It's kinda messy because there are three layers here:
- kola/mantle injecting its own update disablement (https://github.com/coreos/mantle/pull/1018)
- developer builds having updates disabled by default (https://github.com/coreos/fedora-coreos-config/pull/112/commits/8bae3e31cd535ac703cf30cc99768b33e9aba974)
- non-production builds having updates disabled by default (https://github.com/coreos/fedora-coreos-config/pull/112/commits/43d060ac5ac7f9f94170abf524bca077b609898b)

So instead of undoing all of those, this test would basically just add another dropin with an even higher priority. Though maybe it is better indeed test-wise to undo the various "disablements" instead of slapping on a new one that overrides everything else. (Though the first one at least, we should just be able to have the test opt-out from it so we don't have to then undo it.)